app/models/events/comment_announced.rb

```diff
@@ -0,0 +1,29 @@
+class Events::CommentAnnounced < Event
+  include Events::Notify::InApp
+  include Events::Notify::ByEmail
+
+  def self.publish!(model, actor, audiences)
+    super model,
+          user: actor,
+          custom_fields: { audiences: }
+  end
+
+  private
+
+  def email_recipients
+    notification_recipients.where(email_when_mentioned: true)
+  end
+
+  def notification_recipients
+    recipients_ids = custom_fields['audiences'].inject([]) do |ids, audience|
```

Comment:
I know reduce/inject is considered really clever, but I've never seen a use of it that did not take me a very long time to understand. 

To me writing 
```
ids = []
custom_fields['audiences'].each do |audience| 
ids.concat AnnouncementService.audience_users(eventable, audience, eventable.author)
end
```
is a lot easier to read. 

Ether that or how about?

```
custom_fields['audiences'].map do |audience| 
AnnouncementService.audience_users(eventable, audience, eventable.author)
end.flatten.uniq
```

I think I would write the last version if it were me.

Reply:
dont' feel obliged to change this.. I think I might just need to get comfortable with reduce